### PR TITLE
advisor: update api and guide docs of advisor, rename type.ts in ruler as types.ts

### PR DESCRIPTION
**Pull request by GuangMingYouBei**

### CODE REVIEW

File: packages/ava/src/advisor/advise-pipeline/advicesForChart.ts [hunk 1]
```diff
@@ -37,15 +37,23 @@ export function advicesForChart(
...
 
+  const defaultDataProps = dataFrame.info();
   // get dataProps from dataframe
   let dataPropsForAdvice: BasicDataPropertyForAdvice[];
   if (dataProps) {
     // filter out fields that are not included for advising
     dataPropsForAdvice = fields
       ? dataProps.filter((dataProp: BasicDataPropertyForAdvice) => fields.includes(dataProp.name))
       : dataProps;
+    dataPropsForAdvice.map((item) => {
```
Comment (neoddish) [inline packages/ava/src/advisor/advise-pipeline/advicesForChart.ts:48]: what is this for?

and Array.map would not change itself at all

**Review by neoddish — CHANGES_REQUESTED** [review]

File: packages/ava/src/advisor/ruler/types.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
+// eslint-disable-next-line import/no-unresolved
```
Comment (pddpd) [inline packages/ava/src/advisor/ruler/types.ts:1]: is the ignore necessary?

File: packages/ava/src/advisor/ruler/types.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+import type { DataTypes } from '../../data';
+import type { CkbTypes } from '../../ckb';
+import type { Specification } from '../../common/types';
+
```
Comment (pddpd) [inline packages/ava/src/advisor/ruler/types.ts:5]: plz modify this file to `rename` instead of `new file`

File: site/.dumirc.ts [hunk 4]
```diff
@@ -217,9 +217,9 @@ export default defineConfig({
             order: 2,
         },
         {
-            slug: 'chart-advisor',
+            slug: 'advisor',
             title: {
-                zh: '图表推荐（ChartAdvisor）',
+                zh: '图表推荐（Advisor）',
```
Comment (pddpd) [inline site/.dumirc.ts:222]: @LAI-X and u both modify `site`?

File: packages/ava/src/advisor/advise-pipeline/advicesForChart.ts (hunk 1 above)
Comment (GuangMingYouBei) [reply packages/ava/src/advisor/advise-pipeline/advicesForChart.ts]: removed

File: site/.dumirc.ts (hunk 4 above)
Comment (GuangMingYouBei) [reply site/.dumirc.ts:222]: that's reasonable，because she don't know about rename matters.

File: packages/ava/src/advisor/ruler/types.ts (hunk 3 above)
Comment (GuangMingYouBei) [reply packages/ava/src/advisor/ruler/types.ts]: done

File: packages/ava/src/advisor/ruler/types.ts (hunk 2 above)
Comment (GuangMingYouBei) [reply packages/ava/src/advisor/ruler/types.ts]: removed

**Review by neoddish — APPROVED** [review]
lgtm

**Pull request merged**